PetaPoco/Core/PocoData.cs

```diff
@@ -225,9 +225,19 @@ public Delegate GetFactory(string sql, string connectionString, int firstColumn,
                     for (int i = firstColumn; i < firstColumn + countColumns; i++)
                     {
                         // Get the PocoColumn for this db column, ignore if not known
-                        PocoColumn pc;
-                        if (!Columns.TryGetValue(reader.GetName(i), out pc))
-                            continue;
+                        PocoColumn pc;
+                        if (!Columns.TryGetValue(reader.GetName(i), out pc))
+                        {
+                            var k = Columns.FirstOrDefault(x => TableInfo.TableName + "." + x.Key == reader.GetName(i));

```

Comment:
A dictionary TryGetValue approaches an O(1) operation. If the column doesn't exist, we might be adding significant overhead here, as this would be O(N). Obviously, you added it for a reason. Can I ask why you're checking for `KeyValuePair<string, PocoColumn>` in here and the multi poco? I can't see you adding this, so I'm thinking maybe it was missing?

Reply:
AccessDB  in case of joins between tables can return this format
TableName1.exampleName, TableName2.exampleName
and  for example: 
reader.GetName(i) =="TableName1.exampleName" or 
reader.GetName(i) =="TableName2.exampleName"

Reply:
Ah, I see